### PR TITLE
tests: Ignore Not Implemented error for tests

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: functional/FunctionalTest.java [hunk 1]
```diff
@@ -2070,9 +2079,18 @@ public static void setBucketPolicy_test1() throws Exception {
...
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse == null || errorResponse.errorCode() != ErrorCode.NOT_IMPLEMENTED) {
+        mintFailedLog("setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)", null, startTime,
+                      null, e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+        throw e;
+      }
```
Comment (kannappanr) [inline functional/FunctionalTest.java]: Why is the NOT_IMPLEMENTED return code checked only for the case where you are setting the policy type to PolicyType.None, why not in the other scenarios when calling the SetBucketPolicy API?

Comment (balamurugana) [reply functional/FunctionalTest.java]: Done. Fixed.

File: functional/FunctionalTest.java [hunk 2]
```diff
@@ -1994,11 +2004,22 @@ public static void testGetBucketPolicy(String objectPrefix, PolicyType policyTyp
       if (type != policyType) {
         throw new Exception("[FAILED] Expected: " + policyType + ", Got: " + type);
       }
-      mintSuccessLog("getBucketPolicy(String bucketName, String objectPrefix)", null, startTime);
+      mintSuccessLog("" + policyType + ": getBucketPolicy(String bucketName, String objectPrefix)", null, startTime);
```
Comment (nitisht) [inline functional/FunctionalTest.java]: In mint, `function` is logged as a string with the name of function being tested (getBucketPolicy) in this case. Adding `policyType` is not helpful.

File: functional/FunctionalTest.java [hunk 3]
```diff
@@ -1994,11 +2004,22 @@ public static void testGetBucketPolicy(String objectPrefix, PolicyType policyTyp
...
-      throw e;
+      ErrorResponse errorResponse = null;
+      if (e instanceof ErrorResponseException) {
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
+        mintIgnoredLog("" + policyType + ": getBucketPolicy(String bucketName, String objectPrefix)", null, startTime);
```
Comment (nitisht) [inline functional/FunctionalTest.java]: In mint, `function` is logged as a string with the name of function being tested (getBucketPolicy) in this case. Adding policyType is not helpful.

File: functional/FunctionalTest.java [hunk 4]
```diff
@@ -1994,11 +2004,22 @@ public static void testGetBucketPolicy(String objectPrefix, PolicyType policyTyp
...
+      } else {
+        mintFailedLog("" + policyType + ": getBucketPolicy(String bucketName, String objectPrefix)", null, startTime,
```
Comment (nitisht) [inline functional/FunctionalTest.java]: Same as above

File: functional/FunctionalTest.java [hunk 5]
```diff
@@ -2067,12 +2088,24 @@ public static void setBucketPolicy_test1() throws Exception {
       }
 
       client.removeObject(bucketName, objectName);
-      mintSuccessLog("setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)",
+      mintSuccessLog("NONE: setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)",
```
Comment (nitisht) [inline functional/FunctionalTest.java]: Same as above

File: functional/FunctionalTest.java [hunk 6]
```diff
@@ -2067,12 +2088,24 @@ public static void setBucketPolicy_test1() throws Exception {
...
-      throw e;
+      ErrorResponse errorResponse = null;
+      if (e instanceof ErrorResponseException) {
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
+        mintIgnoredLog("NONE: setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)",
```
Comment (nitisht) [inline functional/FunctionalTest.java]: Same as above

File: functional/FunctionalTest.java [hunk 7]
```diff
@@ -2115,11 +2148,24 @@ public static void testSetBucketPolicy(String objectPrefix, PolicyType policyTyp
 
       client.removeObject(bucketName, objectName);
 
-      mintSuccessLog("setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)", 
-                     null, startTime);
+      mintSuccessLog("" + policyType + ": setBucketPolicy(String bucketName, String objectPrefix, "
```
Comment (nitisht) [inline functional/FunctionalTest.java]: Same as above

File: functional/FunctionalTest.java [hunk 8]
```diff
@@ -2115,11 +2148,24 @@ public static void testSetBucketPolicy(String objectPrefix, PolicyType policyTyp
...
-                    null, e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+      ErrorResponse errorResponse = null;
+      if (e instanceof ErrorResponseException) {
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
+        mintIgnoredLog("" + policyType + ": setBucketPolicy(String bucketName, String objectPrefix, "
```
Comment (nitisht) [inline functional/FunctionalTest.java]: Same as above

File: functional/FunctionalTest.java [hunk 9]
```diff
@@ -2115,11 +2148,24 @@ public static void testSetBucketPolicy(String objectPrefix, PolicyType policyTyp
...
+                       + "PolicyType policyType)", null, startTime);
+      } else {
+        mintFailedLog("" + policyType + ": setBucketPolicy(String bucketName, String objectPrefix, "
```
Comment (nitisht) [inline functional/FunctionalTest.java]: Same as above

File: functional/FunctionalTest.java (hunk 2 above)
Comment (balamurugana) [reply functional/FunctionalTest.java]: Could you tell me why it is not useful by seeing four repeated logs with
```
getBucketPolicy(String bucketName, String objectPrefix)
getBucketPolicy(String bucketName, String objectPrefix)
getBucketPolicy(String bucketName, String objectPrefix)
getBucketPolicy(String bucketName, String objectPrefix)
``` ?

Comment (nitisht) [reply functional/FunctionalTest.java]: Yes, as we discussed earlier in the call

- json logs in Mint are for parsing by other tools.
- As the `function` entry in the log is meant to hold the API being tested, it is easy to parse and find out stuff like - which API has how many tests, which API failed, etc. Adding custom prefix will make parsing difficult.

Additionally, I don't think this is helpful as
- `PASS` logs are generally not going to be looked at.
- And if something fails, there is the error field with detailed stack trace with line number and other details.

Comment (balamurugana) [reply functional/FunctionalTest.java]: As per this discussion in slack, I defer this comment

Comment (balamurugana) [reply functional/FunctionalTest.java]: I am relaxing my decision and make things done

**Pull request merged**